iceoryx_posh/include/iceoryx_posh/internal/popo/wait_set.inl

```diff
@@ -317,10 +317,77 @@ inline void WaitSet<Capacity>::acquireNotifications(const WaitFunction& wait) no
     }
     else if (!notificationVector.empty())
     {
-        m_activeNotifications = algorithm::uniqueMergeSortedContainers(notificationVector, m_activeNotifications);
+        uniqueMergeSortedNotificationVector(notificationVector, m_activeNotifications)
+            .and_then([&](auto& mergedNotifications) { m_activeNotifications = mergedNotifications; })
+            .or_else([](auto& error) {
+                if (error == NotificationVectorMergeError::MERGED_VECTOR_OVERFLOW)
```

Comment:
Please remove this, this is unreachable and untestable code. Please stick to the original API of the merge function - it is internal API and does not require an expected.
The contract is that it merges two containers until there is no more space left. Then it truncates